Kitodo/src/test/java/org/kitodo/production/interfaces/activemq/KitodoScriptProcessorIT.java

```diff
@@ -0,0 +1,440 @@
+
+            }
+
+            @Override
+            public void setByte(String name, byte value) throws JMSException {
+
+            }
+
+            @Override
+            public void setShort(String name, short value) throws JMSException {
+
+            }
+
+            @Override
+            public void setChar(String name, char value) throws JMSException {
+
+            }
+
+            @Override
+            public void setInt(String name, int value) throws JMSException {
+
+            }
+
+            @Override
+            public void setLong(String name, long value) throws JMSException {
+
+            }
+
+            @Override
+            public void setFloat(String name, float value) throws JMSException {
+
+            }
+
+            @Override
+            public void setDouble(String name, double value) throws JMSException {
+
+            }
+
+            @Override
+            public void setString(String name, String value) throws JMSException {
+
+            }
+
+            @Override
+            public void setBytes(String name, byte[] value) throws JMSException {
+
+            }
+
+            @Override
+            public void setBytes(String name, byte[] value, int offset, int length) throws JMSException {
+
+            }
+
+            @Override
+            public void setObject(String name, Object value) throws JMSException {
+
+            }
+
+            @Override
+            public boolean itemExists(String name) throws JMSException {
+                return false;
+            }
+        }) {
+            @Override
+            public String getMandatoryString(String s) {
+                return "action:test";
+            }
+
+            @Override
+            public Collection<Integer> getCollectionOfInteger(String s) {
+                return Collections.singletonList(1);
+            }
+        };
+
+        // the object to be tested
+        KitodoScriptProcessor underTest = new KitodoScriptProcessor();
+
+        // organize return of results
+        List<String> scriptResult = new ArrayList<>();
+        List<Process> processesResult = new ArrayList<>();
+        Field serviceField = KitodoScriptProcessor.class.getDeclaredField("kitodoScriptService");
+        serviceField.setAccessible(true);
+        serviceField.set(underTest, new KitodoScriptService() {
+            @Override
+            public void execute(List<Process> processes, String script) {
+                scriptResult.add(script);
+                processesResult.addAll(processes);
+            }
+        });
+
+        // carry out test
+        underTest.process(testData);
+
+        // check results
+        assertEquals("should have passed the script to be executed", scriptResult.get(0), "action:test");
+        assertEquals("should have passed one process", processesResult.size(), 1);
+        assertEquals("should have passed process 1", processesResult.get(0).getId(), (Integer) 1);
```

Comment:
Unneeded cast `(Integer)  1` .

Reply:
The cast is needed, otherwise I get the error: _The method assertEquals(String, Object, Object) is ambiguous for the type KitodoScriptProcessorIT._